source/utilities.cc

```diff
@@ -53,6 +54,222 @@ namespace aspect
+                      for (unsigned int ifield=0; ifield!=list_of_field_names.size(); ++ifield)
+                        {
+                          if (keyword_labels.compare(list_of_field_names[ifield]) == 0)
+                            {
+                              // Throw an error if a match was already found...there can be only one
+                              AssertThrow (std::isnan(x_mapped_values[ifield+1]) == true,
+                                           ExcMessage ("The keyword <"
+                                                       + keyword_labels
+                                                       + "> in "
+                                                       + field_name
+                                                       + " is listed multiple times. "
+                                                       "Check that the field id list is "
+                                                       "the same as what is found in "
+                                                       "Compositional fields subsection "
+                                                       "from `set Names of fields'."
+                                                       "Alternatively, if `set Names of fields' "
+                                                       "is not set, the default names are "
+                                                       "C_1, C_2, ..., C_n."));
+
+                              // We passed...capture the value
+                              x_mapped_values[ifield+1] = (Utilities::string_to_double(split_parts[1]));
+                            }
+                        }
+                    }
+                }
+
+              // Go through one more time to make sure everything was set
+              for (unsigned int ifield=0; ifield!=n_fields_in; ++ifield)
+                {
+                  // Ensure that the special keyword "background" or "bg" does not
+                  // exist when there are the same number of fields as what
+                  // is set in the Compositional fields subsection.
+                  if (ifield == 0)
+                    {
+                      AssertThrow (std::isnan(x_mapped_values[ifield]) == false,
+                                   ExcMessage ("The background field must be set in "
+                                               + field_name
+                                               + ". Set this value with the keyword id "
+                                               "`background' or `bg' with the format "
+                                               "`<id> : <value>'."));
+                    }
+                  else
+                    {
+                      AssertThrow (std::isnan(x_mapped_values[ifield]) == false,
+                                   ExcMessage ("The keyword from <"
+                                               + x_mapped_field_entries[ifield]
```

Comment:
Have you checked if that works? The entries in `x_mapped_field_entries` and `x_mapped_values` don't have to be in the same order, correct? They are in the order given in `list_of_field_names`, so you would have to use `list_of_field_names[ifield-1]` in the error message. Or did I get that wrong?